File: ABSTRACTFACTORY/src/abstractfactory/CriaAnimal.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package abstractfactory;

/**
 *
 * @author Mateus
 */
public class CriaAnimal {

    public Animal criaAnimal(String tipo) {
        switch (tipo) {
            case "Cachorro":
                return new Cachorro();
            case "Papagaio":
                return new Papagaio();
            case "Peixe":
                return new Peixe();
            case "Gato":
                return new Gato();
            default:
                return new Cachorro();
        }
    }
}

File: FACTORYMETHOD/CriaAnimal.php
<?php

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 * Description of CriaAnimal
 *
 * @author Mateus
 */
require_once 'Cachorro.php';
require_once 'Gato.php';
require_once 'Peixe.php';
class CriaAnimal {

    //put your code here
    public function makeAnimal($type) {
        $animal = NULL;
        switch ($type) {
            case "Cachorro":
                $animal = new Cachorro;
                break;
            case "Gato":
                $animal = new Gato;
                break;
            case "Peixe":
                $animal = new Peixe;
                break;
        }
        return $animal;
    }
}
